checker/jdk/nullness/src/org/w3c/dom/Attr.java

```diff
@@ -0,0 +1,307 @@
+ * <pre>
+ * "x=5&amp;#10;y=6"</pre>
+ * </td>
+ * <td valign='top' rowspan='1' colspan='1'>
+ * <pre>"x=5 y=6"</pre>
+ * </td>
+ * <td valign='top' rowspan='1' colspan='1'>
+ * <pre>"x=5&amp;#10;y=6"</pre>
+ * </td>
+ * </tr>
+ * <tr>
+ * <td valign='top' rowspan='1' colspan='1'>Normalized newline between</td>
+ * <td valign='top' rowspan='1' colspan='1'>
+ * <pre>"x=5
+ * y=6"</pre>
+ * </td>
+ * <td valign='top' rowspan='1' colspan='1'>
+ * <pre>"x=5 y=6"</pre>
+ * </td>
+ * <td valign='top' rowspan='1' colspan='1'>
+ * <pre>"x=5 y=6"</pre>
+ * </td>
+ * </tr>
+ * <tr>
+ * <td valign='top' rowspan='1' colspan='1'>Entity <code>e</code> with literal newline</td>
+ * <td valign='top' rowspan='1' colspan='1'>
+ * <pre>
+ * &lt;!ENTITY e '...&amp;#10;...'&gt; [...]&gt; "x=5&amp;e;y=6"</pre>
+ * </td>
+ * <td valign='top' rowspan='1' colspan='1'><em>Dependent on Implementation and Load Options</em></td>
+ * <td valign='top' rowspan='1' colspan='1'><em>Dependent on Implementation and Load/Save Options</em></td>
+ * </tr>
+ * </table>
+ * <p>See also the <a href='http://www.w3.org/TR/2004/REC-DOM-Level-3-Core-20040407'>Document Object Model (DOM) Level 3 Core Specification</a>.
+ */
+public interface Attr extends Node {
+    /**
+     * Returns the name of this attribute. If <code>Node.localName</code> is
+     * different from <code>null</code>, this attribute is a qualified name.
+     */
+    @Pure public String getName();
+
+    /**
+     *  <code>True</code> if this attribute was explicitly given a value in
+     * the instance document, <code>false</code> otherwise. If the
+     * application changed the value of this attribute node (even if it ends
+     * up having the same value as the default value) then it is set to
+     * <code>true</code>. The implementation may handle attributes with
+     * default values from other schemas similarly but applications should
+     * use <code>Document.normalizeDocument()</code> to guarantee this
+     * information is up-to-date.
+     */
+    @Pure public boolean getSpecified();
+
+    /**
+     * On retrieval, the value of the attribute is returned as a string.
+     * Character and general entity references are replaced with their
+     * values. See also the method <code>getAttribute</code> on the
+     * <code>Element</code> interface.
+     * <br>On setting, this creates a <code>Text</code> node with the unparsed
+     * contents of the string, i.e. any characters that an XML processor
+     * would recognize as markup are instead treated as literal text. See
+     * also the method <code>Element.setAttribute()</code>.
+     * <br> Some specialized implementations, such as some [<a href='http://www.w3.org/TR/2003/REC-SVG11-20030114/'>SVG 1.1</a>]
+     * implementations, may do normalization automatically, even after
+     * mutation; in such case, the value on retrieval may differ from the
+     * value on setting.
+     */
+    @Pure public @Nullable String getValue();
```

Comment:
You gave a link to an HTML document.  What does line 209 mean in that document?  Can you give a direct link to the relevant part?  I asked for the exact words that justified your decision, and you didn't give the exact words.

If the value of an attribute is returned as a String, that could mean that for null the string "null" is returned, or it could mean that the null value is returned.  Or, it could mean that the client shouldn't call getValue() on a null attribute.  How do you know which one is the intended specification?

Furthermore, as noted before, even if an attribute can be null in a Node, that is unrelated to the specification of Attr's getValue() routine.